src/ArmTemplates/Commands/Executors/ExtractorExecutor.cs

```diff
@@ -728,6 +734,25 @@ public async Task<Template<IdentityProviderResources>> GenerateIdentityProviderT
             return gatewayApiTemplate;
         }
 
+        public async Task<Template<ApiManagementServiceResources>> GenerateApiManagementServiceTemplate(string baseFilesGenerationDirectory)
+        {
+            var apiManagementServiceTemplate = await this.apiManagementServiceExtractor.GenerateApiManagementServicesTemplateAsync(this.extractorParameters);
+
+            if (apiManagementServiceTemplate?.HasResources() == true)
```

Comment:
A thought - If it's false, you might want to log this so that it's clear there are no resources

Reply:
The extractor will fail in the start if the api management instance was not found, so it would not reach this place